test/framework/machines.go

```diff
@@ -0,0 +1,385 @@
+			return false, nil
+		}
+		return true, nil
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Verify machineset's underlying instances is running")
+	machines, err := f.CAPIClient.ClusterV1alpha1().Machines(machineset.Namespace).List(metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(machineset.Spec.Selector.MatchLabels).String(),
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	for _, machine := range machines.Items {
+		f.waitForMachineToRun(&machine, client)
+	}
+}
+
+func (f *Framework) DeleteMachineSetAndWait(machineset *clusterv1alpha1.MachineSet, client CloudProviderClient) error {
+	By("Get all machineset's machines")
+	machines, err := f.CAPIClient.ClusterV1alpha1().Machines(machineset.Namespace).List(metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(machineset.Spec.Selector.MatchLabels).String(),
+	})
+	if err != nil {
+		return err
+	}
+
+	By(fmt.Sprintf("Deleting %q machineset", machineset.Name))
+	err = f.CAPIClient.ClusterV1alpha1().MachineSets(machineset.Namespace).Delete(machineset.Name, &metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+
+	By("Waiting for all machines to be deleted")
+	for _, machine := range machines.Items {
+		f.waitForMachineToTerminate(&machine, client)
+		// TODO(jchaloup): run it one more time depending on the error returned
+	}
+
+	err = wait.Poll(PollInterval, PoolTimeout, func() (bool, error) {
+		_, err := f.CAPIClient.ClusterV1alpha1().MachineSets(machineset.Namespace).Get(machineset.Name, metav1.GetOptions{})
+		if err != nil {
+			if strings.Contains(err.Error(), "not found") {
+				return true, nil
+			}
+			return false, nil
+		}
+		return true, nil
+	})
+	return err
+}
+
+func (f *Framework) WaitForNodesToGetReady() {
+	err := wait.Poll(PollNodeInterval, PoolNodesReadyTimeout, func() (bool, error) {
+		items, err := f.KubeClient.CoreV1().Nodes().List(metav1.ListOptions{})
+		if err != nil {
+			return false, fmt.Errorf("unable to list nodes: %v", err)
+		}
+		if len(items.Items) < 2 {
+			log.Infof("Waiting for both nodes to come up, have %v", len(items.Items))
+			return false, nil
+		}
+		allNodesReady := true
+		for _, node := range items.Items {
+			for _, condition := range node.Status.Conditions {
+				if condition.Type != apiv1.NodeReady {
+					continue
+				}
+				if condition.Status != apiv1.ConditionTrue {
+					log.Infof("Node %q not ready", node.Name)
+					allNodesReady = false
+				} else {
+					log.Infof("Node %q is ready", node.Name)
+				}
+				break
+			}
+		}
+
+		if !allNodesReady {
+			return false, nil
+		}
+
+		return true, nil
+	})
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func ReadKubeconfigFromServer(sshuser, sshhost, prkey string) (string, error) {
+	fp, err := os.Open(prkey)
+	if err != nil {
+		return "", fmt.Errorf("unable to open id_rsa file: %v", err)
+	}
+	defer fp.Close()
+
+	buf, err := ioutil.ReadAll(fp)
+	if err != nil {
+		return "", fmt.Errorf("unable to read id_rsa file: %v", err)
```

Comment:
Spell out, as suggested above, the filename that cannot be read. This is assuming a default if id_rsa, whereas we pass in a parameter `prkey`.